cpp/include/nebula/client/Connection.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "common/graph/Response.h"
+
+namespace folly {
+    class ScopedEventBaseThread;
+}
+
+namespace nebula {
+
+// Wrap the thrift client.
+namespace graph {
+namespace cpp2 {
+class GraphServiceAsyncClient;
+}
+}
+
+class Connection {
+public:
+    using ExecuteCallback = std::function<void(ExecutionResponse &&)>;
+    using ExecuteJsonCallback = std::function<void(std::string &&)>;
+
+    Connection();
+    // disable copy
+    Connection(const Connection &) = delete;
+    Connection &operator=(const Connection &c) = delete;
+
+    Connection(Connection &&c) noexcept {
+        client_ = c.client_;
+        c.client_ = nullptr;
+
+        clientLoopThread_ = c.clientLoopThread_;
+        c.clientLoopThread_ = nullptr;
+
+        address_ = std::move(c.address_);
+
+        port_ = c.port_;
+        c.port_ = -1;
+    }
+
+    Connection &operator=(Connection &&c);
+
+    ~Connection();
+
+    bool open(const std::string &address, int32_t port);
+    bool open();
```

Comment:
why offer two open, the open() has anything else use?

Reply:
Fixed.